src/Test/QuickCheck/Arbitrary.purs

```diff
@@ -143,3 +146,56 @@ instance arbitraryLazy :: Arbitrary a => Arbitrary (Lazy a) where
 
 instance coarbLazy :: Coarbitrary a => Coarbitrary (Lazy a) where
   coarbitrary a = coarbitrary (force a)
+
+
+instance arbitraryNoArguments :: Arbitrary NoArguments where
+  arbitrary = pure NoArguments
+
+instance coarbitraryNoArguments :: Coarbitrary NoArguments where
+  coarbitrary NoArguments = id
+
+instance arbitrarySum :: (Arbitrary l, Arbitrary r) => Arbitrary (Sum l r) where
+  arbitrary = arbitrary >>= if _ then Inl <$> arbitrary else Inr <$> arbitrary
```

Comment:
I think if we can fix this, we should merge this. Whether we decide to remove `Arbitrary` or not later, we'll need to keep _some_ class to support generic deriving anyway.